crates/router/src/connector/nmi/transformers.rs

```diff
@@ -0,0 +1,773 @@
+    }
+}
+
+impl TryFrom<types::RefundsResponseRouterData<api::Capture, StandardResponse>>
+    for types::RefundsRouterData<api::Capture>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::Capture, StandardResponse>,
+    ) -> Result<Self, Self::Error> {
+        let refund_status = enums::RefundStatus::from(item.response.response);
+        Ok(Self {
+            response: Ok(types::RefundsResponseData {
+                connector_refund_id: item.response.transactionid,
+                refund_status,
+            }),
+            ..item.data
+        })
+    }
+}
+
+impl From<Response> for enums::RefundStatus {
+    fn from(item: Response) -> Self {
+        match item {
+            Response::Approved => Self::Pending,
+            Response::Declined | Response::Error => Self::Failure,
+        }
+    }
+}
+
+impl TryFrom<&types::RefundSyncRouterData> for NmiSyncRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::RefundSyncRouterData) -> Result<Self, Self::Error> {
+        let auth = NmiAuthType::try_from(&item.connector_auth_type)?;
+        let transaction_id = item
+            .request
+            .connector_refund_id
+            .clone()
+            .ok_or(errors::ConnectorError::MissingConnectorRefundID)?;
+
+        Ok(Self {
+            security_key: auth.api_key,
+            transaction_id,
+        })
+    }
+}
+
+impl TryFrom<types::RefundsResponseRouterData<api::RSync, QueryResponse>>
+    for types::RefundsRouterData<api::RSync>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::RSync, QueryResponse>,
+    ) -> Result<Self, Self::Error> {
+        let refund_status = enums::RefundStatus::from(item.response.nm_response.transaction.status);
+        Ok(Self {
+            response: Ok(types::RefundsResponseData {
+                connector_refund_id: item.response.nm_response.transaction.transaction_id,
+                refund_status,
+            }),
+            ..item.data
+        })
+    }
+}
+
+impl From<NmiStatus> for enums::RefundStatus {
+    fn from(item: NmiStatus) -> Self {
+        match item {
+            NmiStatus::Abandoned
+            | NmiStatus::Cancelled
+            | NmiStatus::Failed
+            | NmiStatus::Unknown => Self::Failure,
+            NmiStatus::Pendingsettlement | NmiStatus::Pending | NmiStatus::InProgress => {
+                Self::Pending
+            }
+            NmiStatus::Complete => Self::Success,
+        }
+    }
+}
+
+pub fn get_query_info(
+    query_response: String,
+) -> Result<(String, NmiStatus), errors::ConnectorError> {
+    let mut reader = Reader::from_str(&query_response);
+    reader.trim_text(true);
+
+    let mut buf: Vec<u8> = Vec::new();
+    let mut transaction_id = String::new();
+    let mut condition = String::new();
+
+    loop {
```

Comment:
instead of iterating the xml elements can we parse this string as Struct and access the fields in it? 
Also move the XML parsing logic to crates/common_utils/src/ext_traits.rs